cli/lib/kontena/cli/stacks/stacks_helper.rb

```diff
@@ -52,16 +53,25 @@ def wait_for_deploy_to_finish(deployment, timeout = 600)
         if deployment['state'] == 'error'
           deployment['service_deploys'].each do |service_deploy|
             if service_deploy['state'] == 'error'
-              error service_deploy['reason']
+              errors << pastel.red("#{service_deploy['service_id']} deploy failed: #{service_deploy['reason']}")
+              service_deploy['instance_deploys'].each do |instance_deploy|
+                if instance_deploy['state'] == 'error'
+                  errors << pastel.red(" - #{instance_deploy['error']} on node #{instance_deploy['node']}")
+                end
+              end
             end
           end
-          abort
         end
       end
 
+      unless errors.empty?
```

Comment:
Although sadly there is no stack-level `deployment['error']` message to show in these kinds of cases, it's only in the server logs.

This can happen in e.g. the #2128 case of deploying an upgraded stack after some other service has linked to a service that is being removed.